devito/types/grid.py

```diff
@@ -211,22 +226,35 @@ def spacing(self):
         spacing = (np.array(self.extent) / (np.array(self.shape) - 1)).astype(self.dtype)
         return as_tuple(spacing)
 
-    @property
+    @cached_property
     def spacing_symbols(self):
-        """Symbols representing the grid spacing in each SpaceDimension"""
-        return as_tuple(d.spacing for d in self.dimensions)
+        """Symbols representing the grid spacing in each SpaceDimension."""
+        return as_tuple(d.root.spacing for d in self.dimensions)
 
-    @property
+    @cached_property
     def spacing_map(self):
         """Map between spacing symbols and their values for each SpaceDimension."""
-        return dict(zip(self.spacing_symbols, self.spacing))
+        mapper = {}
+        for d, s in zip(self.dimensions, self.spacing):
+            # Special case subsampling: `Grid.dimensions` -> (xb, yb, zb)` where
+            # `xb, yb, zb` are ConditionalDimensions whose parents are SpaceDimensions
+            if d.is_Conditional:
+                try:
+                    a, b = d.spacing.args
+                    mapper[b] = s/self.dtype(a)
+                    continue
+                except (AttributeError, ValueError):
+                    pass
 
-    @property
-    def origin_offset(self):
-        """Offset of the local (per-process) origin from the domain origin."""
-        grid_origin = [min(i) for i in self.distributor.glb_numb]
-        assert len(grid_origin) == len(self.spacing)
-        return tuple(i*h for i, h in zip(grid_origin, self.spacing))
+            # Typical case: `Grid.dimensions` -> (x, y, z)` where `x, y, z` are
+            # the SpaceDimensions
+            if d.is_Space:
+                mapper[d.spacing] = s
+                continue
+
+            raise RuntimeError("Unable to build `spacing_map` for `%s`" % self)
```

Comment:
Would add something along the lines of `"Unsupported dimension %s type for dimension %s " (d.name, d.type)` for clarity

Reply:
OK, doing it

Reply:
Actually, this should be caught in the constructor, and this exception should be turned into an `assert False` if you see what I mean. Will do this